blscosi/protocol/sub_protocol.go

```diff
@@ -190,36 +190,43 @@ func (p *SubBlsCosi) dispatchRoot() error {
 		return nil
 	}
 
-	// Only one child anyway
-	err := p.SendToChildren(&Announcement{
-		Msg:       p.Msg,
-		Data:      p.Data,
-		Timeout:   p.Timeout,
-		Threshold: p.Threshold,
-	})
-	if err != nil {
-		// Only log what happened so we can try to finish the protocol
-		// (e.g. one child is offline)
-		log.Warnf("Error when broadcasting to children: %s", err.Error())
-	}
+	subLeaderActive := make(chan error, 1)
+	// Because SendToChildren blocks on some firewalls instead of returning
+	// an error, this call is put in a go-routine.
+	go func() {
+		subLeaderActive <- p.SendToChildren(&Announcement{
+			Msg:       p.Msg,
+			Data:      p.Data,
+			Timeout:   p.Timeout,
+			Threshold: p.Threshold,
+		})
+	}()
 
-	select {
-	case <-p.closeChan:
-		return nil
-	case reply := <-p.ChannelResponse:
-		if reply.Equal(p.Root().Children[0]) {
-			// Transfer the response to the parent protocol
-			p.subResponse <- reply
+	for {
+		select {
+		case err := <-subLeaderActive:
+			if err != nil {
+				p.subleaderNotResponding <- true
+				log.Warnf("Couldn't contact subleader: %v", err)
+				return nil
```

Comment:
feels like it should return the `err`

Reply:
I thought that if `Dispatch` returns true, the protocol aborts. But it seems to continue, so that's better.